frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+                            <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${selectId} style='border: 0px !important; font-family: FontAwesome;'>
+                          </div>`;
+          }
+          selectHTML += `<option value='${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T}'>${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T} </option>`;
+        }
+        selectHTML += `</select>`;
+        tableCell.innerHTML = selectHTML;
+      }
+      tableRow.appendChild(tableCell);
+    }
+    tableBody.appendChild(tableRow);
+  }
+  globalDataTypeTable.appendChild(tableBody);
+  globalDataTypeDiv = document.getElementById('globalDataType');
+  globalDataTypeDiv.innerHTML = '';
+  globalDataTypeDiv.appendChild(globalDataTypeTable);
+  tooltipHandler();
+}
+
+/**
+ * Function to update data types with warnings(if any) in global data type table
+ *
+ * @param {string} id id of select box in global data type table
+ * @return {null}
+ */
+const dataTypeUpdate = (id) => {
+  idNum = parseInt(id.match(/\d+/), 10);
+  dataTypeOptionArray = globalDataTypeList[document.getElementById('dataTypeKey' + idNum).innerHTML];
+
+  optionHTML = '';
+  selectHTML = `<div style='display: flex;'>
+                  <i class="large material-icons warning" style='cursor: pointer; visibility: hidden;'>warning</i>
+                  <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                </div>`;
+  for (var x = 0; x < dataTypeOptionArray.length; x++) {
+    optionFound = false;
+    if (dataTypeOptionArray[x].T == document.getElementById(id).value) {
+      optionFound = true;
+    }
+    if (dataTypeOptionArray[x].T == document.getElementById(id).value && dataTypeOptionArray[x].Brief != "") {
+
+      selectHTML = `<div style='display: flex;'>
+                      <i class="large material-icons warning" style='cursor: pointer;' data-toggle='tooltip' data-placement='bottom' title='${dataTypeOptionArray[x].Brief}'>warning</i>
+                      <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                    </div>`;
+    }
+    if (optionFound == true) {
+      optionHTML += `<option selected='selected' value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+    else {
+      optionHTML += `<option value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+  }
+  selectHTML += optionHTML + `</select>`;
+  document.getElementById('dataTypeVal' + idNum).innerHTML = selectHTML;
+  tooltipHandler();
+}
+
+/**
+ * Function to create table from con json structure
+ *
+ * @param {json} obj Json object contaning source and spanner table information
+ * @return {null}
+ */
+async function createSourceAndSpannerTables(obj) {
```

Comment:
Changed the variables scope inside createSourceAndSpannerTables and other functions

Reply:
@13shreya still there are variables used that are not declared anywhere for ex.
schemaConversionObj, interleaveApiCallResp

Reply:
I have changed scope of all variables in code.. only three global variables are there inside createSourceAndSpannerTable() which are schemaConversionObj, srcTableName and notNullConstraint. I have made these global because we are using srcTableName array in creatDdlFromJson() and createSummaryFromJson() as well and these functions are not dependent on each other and called independently.. schemaConversionObj variable is used in schema-conversion.js file also.. that's why made it global

Reply:
Okay these three variables need to be defined somewhere globally then, may be in utility.js file
```
schemaConversionObj, srcTableName, notNullConstraint;
```

Reply:
variables are moved to utility.js file